python/lib/dependabot/python/file_parser.rb

```diff
@@ -91,41 +91,109 @@ def package_manager
 
       sig { returns(Ecosystem::VersionManager) }
       def detected_package_manager
-        return PeotryPackageManager.new(detect_poetry_version) if poetry_lock && detect_poetry_version
+        setup_python_environment if Dependabot::Experiments.enabled?(:enable_file_parser_python_local)
+
+        return PoetryPackageManager.new(detect_poetry_version) if poetry_files && detect_poetry_version
+
+        return PipCompilePackageManager.new(detect_pipcompile_version) if pip_compile_files && detect_pipcompile_version
+
+        return PipenvPackageManager.new(detect_pipenv_version) if pipenv_files && detect_pipenv_version
 
         PipPackageManager.new(detect_pip_version)
```

Comment:
I am pretty sure there are python package managers we don't support yet, I think this should still follow the same logic and only return pip if pip is actually found. This method should return `nil` if no supported package managers are found.

Reply:
> I am pretty sure there are python package managers we don't support yet, I think this should still follow the same logic and only return pip if pip is actually found. This method should return `nil` if no supported package managers are found.

@pavera 
When you return `ecosystem` object in `file_parser`, currently `package_manager` is required field. If we are unable detect `package_manager` I believe we shouldn't be able to hit this method since this method is getting called before starting update operation after parsings file. In this situation we should return error or something for not supported package manager. 

@sachin-sandhu 
Is there a way to be sure that we are returning the right package manager?

Reply:
@kbukum1 , @pavera , right now PIP (only) is a fallback package manager, if we are not able to meet any conditions for a rest of package managers (poetry, pip-compile, pipenv), we take PIP as the last resorting manager. 

https://github.com/dependabot/dependabot-core/blob/e0c9aa891f81a6bddc32a3ab7f1bddac80200702/python/lib/dependabot/python/update_checker.rb#L125C1-L144C10

Reply:
@kbukum1 , @pavera , right now PIP (only) is a fallback package manager, if we are not able to meet any conditions for a rest of package managers (poetry, pip-compile, pipenv), we take PIP as the last resorting manager. there is no exclusive condition to check for PIP other than negation of other package managers.

https://github.com/dependabot/dependabot-core/blob/e0c9aa891f81a6bddc32a3ab7f1bddac80200702/python/lib/dependabot/python/update_checker.rb#L142